test/inductor/test_torchinductor.py

```diff
@@ -7712,6 +7712,48 @@ def fn(pytype, dtype):
 
                 self.assertEqual(ret_opt, fn(pytype, dtype))
 
+        # Test that the compiler and torch output for binary operations match.
+        # Particularly check scenarios where the operation performed would result in 'wrapping',
+        # i.e., when the result would be too big for the type so a negative value is returned.
+        def test_lowering(self):
+
+            def mySum16(x):
+                return (x + x).to(torch.int16)
+            def myMul16(x):
+                return (x * x).to(torch.int16)
+            def mySquare16(x):
+                return (x ** 2).to(torch.int16)
+
+            x = torch.tensor(128, dtype=torch.uint8)
+
+            torchResult = mySum16(x)
+            dynamoResult = torch.compile(mySum16)(x)
+
+            assert(torchResult == dynamoResult == 0)
```

Comment:
Rather than doing this, see `self.common`, which also provides a few other tests behind the hood.

Reply:
> Rather than doing this, see `self.common`, which also provides a few other tests behind the hood.

@lezcano I have had a look at this. I see there is a common template that is used to run CPU and CUDA tests.
I have a mac system, this `if` does not run as  torch.backends.mps.is_available() is true. Commenting out the mps check gives me errors.

```
 if HAS_CPU and not torch.backends.mps.is_available():

    class SweepInputsCpuTest(SweepInputs2, TestCase):
        gen = InputGen(10, "cpu")

    SweepInputsCpuTest.populate()

    class CpuTests(TestCase):
        common = check_model
        device = "cpu"


    copy_tests(CommonTemplate, CpuTests, "cpu") 

```
I will make the changes and see if the pr build works with the tests in the new place

Reply:
the tests have all passed in the template place :-)